### ISSUE TITLE
S3*: Rules now also report non-function calls

### ISSUE THREAD

**Issue by MichaReiser** [astral-sh/ruff#16673]
### Summary

https://github.com/astral-sh/ruff/pull/15541 changed the S3* rules to report all references instead of only checking for the problematic functions in call expressions (callee) positions to detect them when used in callback positions. 

I think there are two issues with the change: 

1. The rule uses the same error message regardless if the function is used in a call position or not: *S321 FTP-related functions are being called. FTP is considered insecure. Use SSH/SFTP/SCP or some other encrypted protocol.* The `are being called` part can be confusing when there's no actual call, e.g. `port = ftplib.FTP_PORT`
2. The rule now also reports usages for symbols that aren't functions
  ```py
          if self.conn is None:
            params = self.get_connection(self.ftp_conn_id)
            pasv = params.extra_dejson.get("passive", True)
            self.conn = ftplib.FTP()  # nosec: B321
            if params.host:
                port = ftplib.FTP_PORT
                if params.port is not None:
                    port = params.port
                logger.info("Connecting via FTP to %s:%d", params.host, port)
                self.conn.connect(params.host, port)
                if params.login:
                    self.conn.login(params.login, params.password)
            self.conn.set_pasv(pasv)
  ```

Some of this could be addressed by rephrasing the message or we want a more sophisticated check to e.g. only check for usages in function call arguments rather than references anywhere in code.

**Comment by statxc** [comment]
@MichaReiser Can I work on this? thanks
